packages/example-project/component-library-react/__tests__/my-component.spec.tsx

```diff
@@ -11,25 +11,45 @@ describe('MyComponent', () => {
     expect(component).toBeInTheDocument();
   });
 
-  it('should get strings as props', () => {
+  it('should render attributes as dash-case', () => {
+    const { webcomponent: myComponent } = includeWebComponent<HTMLMyComponentElement>(
+      renderWithStrictMode(<MyButton buttonType="my-button-type" />),
+    );
+    expect(myComponent.hasAttribute('buttonType')).toBe(false);
+    expect(myComponent.getAttribute('button-type')).toBe('my-button-type');
+  });
+
+  it('should get strings as props and attrs', () => {
     const { webcomponent: myComponent } = includeWebComponent<HTMLMyComponentElement>(
       renderWithStrictMode(<MyComponent first="blue" />),
     );
     expect(myComponent.first).toEqual('blue');
+    expect(myComponent.getAttribute('first')).toBe('blue');
   });
 
-  it('should get numbers as props', () => {
+  it('should get numbers as props and attrs', () => {
     const { webcomponent: myComponent } = includeWebComponent<HTMLMyComponentElement>(
       renderWithStrictMode(<MyComponent age={39} />),
     );
     expect(myComponent.age).toEqual(39);
+    expect(myComponent.getAttribute('age')).toBe('39');
+  });
+
+  it('should get booleans as props and attrs', () => {
+    const { webcomponent: myComponent } = includeWebComponent<HTMLMyInputElement>(
+      renderWithStrictMode(<MyInput clearInput />),
+    );
+    expect(myComponent.clearInput).toEqual(true);
+    expect(myComponent.getAttribute('clear-input')).toBe('true');
```

Comment:
Boolean attributes should have an empty string as a value, not 'true'
https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes#boolean_attributes

Reply:
I think Stencil might diverge from the spec here. If you set `some-bool="false" ` it will set the underlying value to false IIRC. Perhaps I can be more careful though?

Reply:
Hmm, I didn't notice it before, but you are right. That's definitely a bug (or a decision on their part).
If you use the prop `elem.someBool = false` it will correctly remove the attribute, instead of setting its value to `"false"`.

Reply:
Perhaps just adding or removing the attr is best. As I said below the aim here is to improve SSR and maximise the amount of data passed via plain ol html

Reply:
One difficulty: booleans props that default to `true` - the only way to set to false is to explicitly pass false

Reply:
The docs show the explicitly passing true/false approach for boolean attributes (I guess they modelled them on aria attributes, which are not like boolean html attributes). So I would say this is intended, not a bug https://stenciljs.com/docs/properties

Reply:
i think that whoever wrote that part of the docs didn't know or didn't realise he was making a mistake.
they are also showing that u can set a string `false` via JSX, which will result in a type error:
![image](https://user-images.githubusercontent.com/66014157/122686141-675b0b00-d218-11eb-9081-f43e257a9767.png)

Reply:
I think that's because you're setting a property in jsx, whereas here we are dealing with attributes in html (see also: difference between dash case attributes in html and camel case props in jsx). I would lean towards this being intentional, without this there would be no way to set a prop which defaults to true to false in html.

Would be useful for one of the Stencil team to chime in here :)

Reply:
I see what you mean about a mistake in the docs though, as they show the string usage even in jsx! I think the html example is correct, the jsx one is not

Reply:
Just stumbled onto this discussion, and wanted to mention that there's an existing issue (#123) requesting that boolean attributes be omitted from a components' rendered HTML when passed `false`, instead of ending up as the string value `"false"`, as they do now.

Personally, I think it makes sense for Stencil's React wrapper to treat boolean attributes the way React itself treats built-in HTML tags like `input`, where passing `false` removes the attribute.

 Unfortunately we have not heard from anyone on the Stencil team on that thread either.

Reply:
Hey folks 👋 

First, I apologize for the delay here.  I'm actively looking into #123 and this PR right now.

Second, I've put up a [small PR](https://github.com/ionic-team/stencil-site/pull/826/files) for fixing the docs as pointed out above.  I'm rewriting the `Props` page almost entirely to make it more comprehensive and a little easier to understand. In the mean time, the PR I linked is hopefully a 'quick win' to show folks how you're supposed to use boolean props in a JSX context.

Thirdly, @WickyNilliams can you explain [this comment](https://github.com/ionic-team/stencil-ds-output-targets/pull/149/files#r654983158) to me? I'm not sure I understand the context:

> without this there would be no way to set a prop which defaults to true to false in html.

Are you talking about setting props in a React/JSX context, via HTML? I want to make sure I understand this point so I can take it into consideration.


Thanks!

Reply:
@rwaskiewicz if you see my [comment above](https://github.com/ionic-team/stencil-ds-output-targets/pull/149#discussion_r653941740), I explain how the HTML spec defines boolean attributes, which to summarise briefly: `If a boolean attribute is present, its value is true, and if it's absent, its value is false.`

What @WickyNilliams  is pointing out is the case that if a property was set with a default value of `true`, which is something we allow in Stencil, if we follow the spec, we have no way of serialising when the value is set to`false`. The HTML spec probably ignored this and made sure that all HTML elements had props that default to `false`, however this is not something we can guarantee in Stencil.

However, I propose an alternative: we simply assume `default true === true` and `default false === false`, i.e:
- When serializing:
  - We always `include` the attribute when the prop is either `true` or `default true`
  - We always `remove` the attribute when the prop is either `false` or `default false`
- When deserializing:
  - We always treat the existence of the attribute as `true` and set the prop to `true`
  - We always treat the absence of the attribute as `false` and set the prop to `false`

This way we will follow the spec, while allowing for both `default true` and `default false` props to work.